File: 프로그래머스/0/181857. 배열의 길이를 2의 거듭제곱으로 만들기/배열의 길이를 2의 거듭제곱으로 만들기.cpp
#include <string>
#include <vector>
#include <math.h>

using namespace std;

vector<int> solution(vector<int> arr) {
    vector<int> answer;
    
    int size = arr.size();
    int next_power_of_two = 1;
    while (next_power_of_two < size) {
        next_power_of_two *= 2;
    }

    int zeros_to_add = next_power_of_two - size;

    for (int i = 0; i < zeros_to_add; ++i) {
        arr.push_back(0);
    }

    answer=arr;
    return answer;
}

File: 프로그래머스/0/181857. 배열의 길이를 2의 거듭제곱으로 만들기/배열의 길이를 2의 거듭제곱으로 만들기.py
def solution(arr):
    # 현재 배열의 길이
    n = len(arr)
    
    # 현재 길이보다 크거나 같은 가장 작은 2의 거듭제곱 찾기
    power_of_two = 1
    while power_of_two < n:
        power_of_two *= 2
    
    # 부족한 길이만큼 0을 추가
    while len(arr) < power_of_two:
        arr.append(0)
    
    return arr
